security/pkg/pki/ca/selfsignedcarootcertrotator.go

```diff
@@ -175,6 +181,10 @@ func (rotator *SelfSignedCARootCertRotator) checkAndRotateRootCertForSigningCert
 		RSAKeySize:    caKeySize,
 		IsDualUse:     rotator.config.dualUse,
 	}
+	// options should be consistent with the one used in NewSelfSignedIstioCAOptions().
+	// This is to make sure when rotate the root cert, we don't make unnecessary changes
+	// to the certificate or add extra fields to the certificate.
+	options = util.MergeCertOptions(options, oldCertOptions)
```

Comment:
Can we add tests for this function as well? Thanks!

Reply:
I have added test TestMergeCertOptions().

Reply:
Sorry, I meant can we test the added code in this checkAndRotateRootCertForSigningCertCitadel function, in selfsignedcarootcertrotator_test.go?

Reply:
I have added a test TestRootCertRotatorKeepCertFieldsUnchanged() into selfsignedcarootcertrotator_test.go, please take a look.